users/vieting/experiments/switchboard/hybrid/feat/specaug_random.py

```diff
@@ -0,0 +1,125 @@
+
+def _mask(x, batch_axis, axis, pos, max_amount, shuffled=False):
+    """
+    :param tf.Tensor x: (batch,time,feature)
+    :param int batch_axis:
+    :param int axis:
+    :param tf.Tensor pos: (batch,)
+    :param int|tf.Tensor max_amount: inclusive
+    """
+    from returnn.tf.compat import v1 as tf
+
+    ndim = x.get_shape().ndims
+    n_batch = tf.shape(x)[batch_axis]
+    dim = tf.shape(x)[axis]
+    amount = tf.random_uniform(shape=(n_batch,), minval=1, maxval=max_amount + 1, dtype=tf.int32)
+    pos2 = tf.minimum(pos + amount, dim)
+    idxs = tf.expand_dims(tf.range(0, dim), 0)  # (1,dim)
+    pos_bc = tf.expand_dims(pos, 1)  # (batch,1)
+    pos2_bc = tf.expand_dims(pos2, 1)  # (batch,1)
+    cond = tf.logical_and(tf.greater_equal(idxs, pos_bc), tf.less(idxs, pos2_bc))  # (batch,dim)
+    if shuffled:
+            cond = tf.transpose(cond)  # (dim,batch)
+            cond = tf.random.shuffle(cond)
+            cond = tf.transpose(cond)  # (batch,dim)
+    if batch_axis > axis:
+        cond = tf.transpose(cond)  # (dim,batch) 
+    cond = tf.reshape(cond, [tf.shape(x)[i] if i in (batch_axis, axis) else 1 for i in range(ndim)])
+    from TFUtil import where_bc
+
+    x = where_bc(cond, 0.0, x)
+    return x
+
+
+def _random_mask(x, batch_axis, axis, min_num, max_num, max_dims, shuffeling):
```

Comment:
```suggestion
def _random_mask(x, batch_axis, axis, min_num, max_num, max_dims, shuffling):
```
Or maybe `shuffled` to stay consistent with the name above.